Assets/SEE/UI/Window/ConsoleWindow/ConsoleWindow.cs

```diff
@@ -0,0 +1,691 @@
+            {
+                item.SetActive(false);
+            }
+        }
+
+        /// <summary>
+        /// Shows the search options.
+        /// </summary>
+        /// <param name="refresh"></param>
+        private void ShowSearchOptionsPopup(bool refresh = false)
+        {
+            popupMenu.ClearEntries();
+
+            popupMenu.AddEntry(new PopupMenuAction("Match Case", () =>
+            {
+                matchCase = !matchCase;
+                ShowSearchOptionsPopup(true);
+                UpdateFilters();
+            }, matchCase ? Icons.CheckedCheckbox : Icons.EmptyCheckbox, false));
+            popupMenu.AddEntry(new PopupMenuAction("Full Match", () =>
+            {
+                fullMatch = !fullMatch;
+                ShowSearchOptionsPopup(true);
+                UpdateFilters();
+            }, fullMatch ? Icons.CheckedCheckbox : Icons.EmptyCheckbox, false));
+
+            if (!refresh)
+            {
+                popupMenu.MoveTo(searchOptionsButton.transform.position);
+                popupMenu.ShowMenuAsync().Forget();
+            }
+        }
+
+        /// <summary>
+        /// Shows the filter options.
+        /// </summary>
+        /// <param name="refresh"></param>
+        private void ShowFilterPopup(bool refresh = false)
+        {
+            popupMenu.ClearEntries();
+
+            foreach (Channel channel in channels.Values)
+            {
+                popupMenu.AddEntry(new PopupMenuHeading(channel.Name));
+                foreach (ChannelLevel level in channel.Levels.Values)
+                {
+                    popupMenu.AddEntry(new PopupMenuAction(level.Name, () =>
+                    {
+                        level.enabled = !level.enabled;
+                        UpdateFilters();
+                        ShowFilterPopup(true);
+                    }, level.enabled ? Icons.CheckedCheckbox : Icons.EmptyCheckbox, false));
+                }
+            }
+            if (!refresh)
+            {
+                popupMenu.MoveTo(filterButton.transform.position);
+                popupMenu.ShowMenuAsync().Forget();
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void RebuildLayout()
+        {
+            // Nothing needs to be done
+        }
+
+        /// <inheritdoc/>
+        protected override void InitializeFromValueObject(WindowValues valueObject)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        /// <inheritdoc/>
+        public override void UpdateFromNetworkValueObject(WindowValues valueObject)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        /// <inheritdoc/>
+        public override WindowValues ToValueObject()
+        {
+            throw new System.NotImplementedException();
+        }
+
+        /// <summary>
+        /// Container for a console message.
+        /// </summary>
+        private class Message
```

Comment:
Consider using a record for a value class like this (i.e., `private record Message(string Channel, string Level, string Text);`)